e2e/logic/api/apiCalls.ts

```
@@ -100,12 +105,86 @@ export default class ApiCalls {
-            return await result.json();
+    
+            let result = await getRequest(`${urls.api.graphUrl}${graphName}?query=${encodeURIComponent(query)}`, headers);
+            let rawText = await result.text();
+
+            let json = JSON.parse(rawText);
+    
+            // Poll if response contains a numeric result (job ID)
+            const MAX_POLLS = 10;
+            let polls = 0;
+            while (typeof json.result === "number") {
+                if (++polls > MAX_POLLS) {
```

ESLint

[error] Unary operator '++' used.

(no-plusplus)